### PR TITLE
Portfolio and Public Editors

### PR DESCRIPTION
**Pull request by kevincharles**

### CODE REVIEW

File: cypress/e2e/Portfolio/ShareActivities.cy.js
Comment (jaltekruse) [inline cypress/e2e/Portfolio/ShareActivities.cy.js:29]: reminder to remove the .only before merging

File: cypress/e2e/Portfolio/ShareActivities.cy.js
Comment (jaltekruse) [inline cypress/e2e/Portfolio/ShareActivities.cy.js:145]: gentle reminder to delete commented out code we don't want before merging

File: public/api/activityThumbnailUpload.php
Comment (jaltekruse) [inline public/api/activityThumbnailUpload.php:137]: It looks like this this todo is already complete above

File: public/api/activityThumbnailUpload.php
Comment (jaltekruse) [inline public/api/activityThumbnailUpload.php:1]: Could be shortened with the new try/catch pattern

File: public/api/supportFileUpload.php
Comment (jaltekruse) [inline public/api/supportFileUpload.php:39]: There is a lot of shared code with the thumbnail API, I know that has the additional restriction of only one per doenetID, but it would be good to figure out how to share the common code between the two.

File: public/api/supportFileUpload.php
Comment (jaltekruse) [inline public/api/supportFileUpload.php:118]: only one of these stores into this ipfs table, ipfs is still disabled, but if we're keeping it around at all for possibly re-enabling, do we not want the thumbnails in IPFS?

File: public/api/upload.php
Comment (jaltekruse) [inline public/api/upload.php:18]: Do we need both this API and the dedicated one for thumbnails?

Comment (jaltekruse) [reply public/api/upload.php:18]: Taking a look at the APIs used, it looks like the two dedicated new ones are used, and there is an unchanged use of this one, but no one is taking advantage of this new feature in this API.

File: cypress/e2e/Portfolio/Load&SaveEditor.cy.js
Comment (jaltekruse) [inline cypress/e2e/Portfolio/Load&SaveEditor.cy.js:5]: These test descriptions should be updated, also note from @dqnykamp , he forgot to do this in ShareActivities, so we should also change that one (you didn't edit close enough to that line on the file for github to let me make a comment there)

File: public/api/activityThumbnailUpload.php
Comment (kevincharles) [reply public/api/activityThumbnailUpload.php:137]: updated

File: public/api/supportFileUpload.php
Comment (kevincharles) [reply public/api/supportFileUpload.php:118]: I'm leaving it for now. ty

File: cypress/e2e/Portfolio/Load&SaveEditor.cy.js
Comment (kevincharles) [reply cypress/e2e/Portfolio/Load&SaveEditor.cy.js:5]: This file has changed.  I'm not sure if this is already fixed.

**Review by jaltekruse — APPROVED** [review]
+1   I'd like to go through these changes a little more together to clean things up after the workshop, but I don't think it needs to block merging this

**Pull request merged**
